[Jeremy Toh] iP

deepimpactmir opened the pull request:
Primer

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/deepimpactmir/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your task for you :wink: 

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Sort your deadlines/events chronologically.

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

deepimpactmir committed:
Level 1 of Duke

deepimpactmir committed:
Level-2

deepimpactmir committed:
Refactored main in Duke and completed Level-3

deepimpactmir committed:
Level-3 with some refactoring

deepimpactmir committed:
Level-4 Corrected

deepimpactmir committed:
Added Automated Text UI Testing

deepimpactmir committed:
Level 5 Minimal

deepimpactmir committed:
Naive implementation of Level-6

deepimpactmir committed:
Handle more exceptions

deepimpactmir committed:
Add new test cases

deepimpactmir committed:
Add capability to save and load tasks

deepimpactmir committed:
Add datetime functionality

deepimpactmir committed:
Merge branch 'level7'

Add save and load functionality

deepimpactmir committed:
Merge branch 'level8'

deepimpactmir committed:
Add Ui to deal with interactions with the user

deepimpactmir committed:
Rework Ui and clean up main in Duke

deepimpactmir committed:
Add Parser to handle user input

deepimpactmir committed:
Add TaskList class for more OOP

deepimpactmir committed:
Package files into duke and duke.task

deepimpactmir committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Add the Gradle wrapper files to the project.

deepimpactmir committed:
Merge branch 'master' of github.com:deepimpactmir/ip

deepimpactmir committed:
Add javadoc for documentation

deepimpactmir committed:
Set up gradle to build and run Duke

deepimpactmir committed:
Modify script for text-ui-test

deepimpactmir committed:
Add JUnit tests for Todo

Minimal implementation for JUnit.

deepimpactmir committed:
Add find command in Parser

Add findTask method in TaskList that returns a new task list of the
filitered tasks. Add some methods in Ui to support the printing of the
matched tasks.

deepimpactmir committed:
Edit to fit coding standard

deepimpactmir committed:
Merge tag 'A-JavaDoc'

Add documentation in the form of javadoc

deepimpactmir committed:
Merge tag 'A-CodingStandard'

Edit files to fit coding standard

deepimpactmir committed:
Merge tag 'Level-9'

Level-9

deepimpactmir committed:
Upgrade Gradle

deepimpactmir committed:
Add Command and its subclass.

deepimpactmir committed:
Add GUI to Duke.

albertarielw commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,19 @@
+package duke.command;
```
First of all, excellent job! I really like this abstraction of command as a class, it makes the main code much neater.

albertarielw commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,24 @@
+import duke.task.Todo;
+
+public class TodoCommand implements Command {
+    public static final String COMMAND_WORD = "todo";
```
Perhaps you can consider extracting out the string so it is easier to edit, but aside from that i like how your code is very neat

albertarielw commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,44 @@
+package duke.task;
```
Event and Deadline share a lot of similarities, perhaps it will make it simpler if they Deadline inherits from Event?

albertarielw commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,91 @@
+ * Stores the description of a task and whether it is done.
+ */
+public abstract class Task {
+  private static final String DONE = "X";
```
I like how you extracted out the string here! One suggestion is using enum or put all the string values in UI and make it a static variables?

albertarielw commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+package duke.task;
```
I really like your detailed javadocs here!

albertarielw commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,23 @@
+todo
```
I like how your test is very detailed also!

jsincorporated replied:
Agreed with the above comment! Very extensive testing✅

deepimpactmir committed:
Merge branch 'branch-Level-10'

Add GUI using javafx

deepimpactmir committed:
Removed unused methods in Ui

deepimpactmir committed:
Refactor the code and add checkstyle in gradle

Change methods in Ui to be static since we technically not longer have
to initialise Ui since we moved on to GUI.

Change the behaviour of the help command, no longer gives usage of all
the commands but instead gives the message for each corresponding
command.

Add checkstyle in gradle

deepimpactmir committed:
Add checkstyle xml for gradle

deepimpactmir committed:
Fix logic of the delete command

deepimpactmir committed:
Add assertions wherever possible

Currently there is no assertions used and it is good practice to use
assertions wherever possible.

Let's
* add assertions to Storage to ensure that the file should
in fact exist before writing or reading.
* add assertions to FindCommand to ensure that the filtered task list
  returned is not null

deepimpactmir committed:
Merge pull request #1 from deepimpactmir/branch-A-Assertions

deepimpactmir committed:
Fix Checkstyles errors

jsincorporated commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * A GUI for Duke using FXML.
+ */
+public class Main extends Application {
```
Nice job implementing a GUI using javaFX👍

jsincorporated reviewed (commented):
Overall, in terms of code quality, I found that it was pretty well done. There are just some individual lines where I feel breaking up the logic would help in readability and understanding of the code. Other than that, I feel that the code quality is quite solid. Great job!

deepimpactmir committed:
Add sort command to Duke

No way for users to sort deadlines.

Users should be able to sort deadlines to see which ones that should be
prioritised.

Let's,
* Add a new method in TaskList to handle the logic for sorting deadlines
* Have DateTime implement Comparable to help with sorting
* Add SortCommand to handle the new command
* Update Parser to include the new sort command

deepimpactmir committed:
Merge branch 'branch-C-sort'

Merge the new changes to introduce sort, which will sort deadlines in
the task list.

deepimpactmir committed:
Refactor Storage class

Changed void functions to return objects whenever possible and changed
some functions that are not accessed from outside of the class to
private.

deepimpactmir committed:
Add CI using Github Actions

deepimpactmir committed:
Change help command message

deepimpactmir committed:
Merge branch 'branch-A-CodeQuality'

Rework Storage and some other minor edits

deepimpactmir committed:
Move the regex strings into the commands itself

deepimpactmir committed:
Merge branch 'branch-A-CodeQuality'

Move the regex strings into the commands

deepimpactmir committed:
Add padding to GUI and display task list on start

deepimpactmir committed:
Round ImageView

deepimpactmir committed:
Fix logic of help bye.

deepimpactmir committed:
Fix javadoc comments for methods in Storage

deepimpactmir committed:
Update README.md

deepimpactmir committed:
Fix wrap text in Label.

deepimpactmir committed:
Remove IO redirection test for Github actions

deepimpactmir committed:
Update README.md

Fix broken page.

deepimpactmir committed:
Update README.md

deepimpactmir committed:
Fix responses of some commands

Fix responses in mark, unmark and help sort.

deepimpactmir committed:
Add a new test for sort in TaskList

deepimpactmir committed:
Make changes to TaskList

To check if there is no tasks in the task list and return the
appropriate string.

deepimpactmir committed:
Fix cross-platform compatibility.

deepimpactmir committed:
Replace Ui.png with the correct image

deepimpactmir committed:
Fix javadoc for methods to adhere to standards
